rskj-core/src/test/java/co/rsk/peg/whitelist/WhitelistStorageProviderImplTest.java

```diff
@@ -0,0 +1,216 @@
+        assertFalse(secondTimeQuery.isWhitelisted(firstBtcAddress));
+
+        //Recreating whitelistStorageProvider to make sure it is querying the storage
+        whitelistStorageProvider = new WhitelistStorageProviderImpl(inMemoryStorage);
+        LockWhitelist actualLockWhitelist = whitelistStorageProvider.getLockWhitelist(activationConfig, networkParameters);
+
+        //Return one entry that was saved directly in storage
+        assertEquals(1, actualLockWhitelist.getAll().size());
+        //Making sure the correct value was whitelisted
+        assertTrue(actualLockWhitelist.isWhitelisted(firstBtcAddress));
+    }
+
+    @Test
+    void getWhitelist_whenIsActiveRSKIP87_shouldReturnUnlimitedEntries() {
+        when(activationConfig.isActive(ConsensusRule.RSKIP87)).thenReturn(true);
+        saveInMemoryStorageOneOffWhiteListEntry(firstBtcAddress);
+        saveInMemoryStorageUnlimitedWhiteListEntry(secondBtcAddress);
+
+        LockWhitelist actualLockWhitelist = whitelistStorageProvider.getLockWhitelist(activationConfig, networkParameters);
+
+        assertEquals(2, actualLockWhitelist.getAll().size());
+        //Making sure the correct value was whitelisted
+        assertTrue(actualLockWhitelist.isWhitelisted(firstBtcAddress));
+        assertTrue(actualLockWhitelist.isWhitelisted(secondBtcAddress));
+    }
+
+    private void saveInMemoryStorageOneOffWhiteListEntry(Address btcAddress) {
+        OneOffWhiteListEntry oneOffWhiteListEntry = createOneOffWhiteListEntry(btcAddress);
+        List<OneOffWhiteListEntry> oneOffWhiteListEntries = Collections.singletonList(oneOffWhiteListEntry);
+        Pair<List<OneOffWhiteListEntry>, Integer> pairValue = Pair.of(oneOffWhiteListEntries, 100);
+
+        inMemoryStorage.safeSaveToRepository(
+            LOCK_ONE_OFF.getKey(),
+            pairValue,
+            BridgeSerializationUtils::serializeOneOffLockWhitelist
+        );
+    }
+
+    private void saveInMemoryStorageUnlimitedWhiteListEntry(Address btcAddress) {
+        UnlimitedWhiteListEntry unlimitedWhiteListEntry = createUnlimitedWhiteListEntry(btcAddress);
+        List<UnlimitedWhiteListEntry> unlimitedWhiteListEntries = Collections.singletonList(unlimitedWhiteListEntry);
+
+        inMemoryStorage.safeSaveToRepository(
+            WhitelistStorageIndexKey.LOCK_UNLIMITED.getKey(),
+            unlimitedWhiteListEntries,
+            BridgeSerializationUtils::serializeUnlimitedLockWhitelist
+        );
+    }
+
+    private OneOffWhiteListEntry createOneOffWhiteListEntry(Address btcAddress) {
+        Coin maxTransferValue = Coin.COIN;
+        return new OneOffWhiteListEntry(btcAddress, maxTransferValue);
+    }
+
+    private UnlimitedWhiteListEntry createUnlimitedWhiteListEntry(Address btcAddress) {
+        return new UnlimitedWhiteListEntry(btcAddress);
+    }
+
+    private Map<Address, OneOffWhiteListEntry> getAddressFromOneOffWhiteListEntryInMemoryStorage() {
```

Comment:
no need to add the `WhiteList` part

Reply:
Agree with @julia-zack

Reply:
The sense is to keep the complete name of the object we are getting, `OneOffWhiteListEntry`. Agree regarding removing `InMemory` from the name. I propose you this name: `getAddressFromOneOffWhiteListEntryStorage`. Please let me know your insights!

Reply:
we are in the whitelist context, there's no need to have the _whitelist_ part.
And i think saying "storage entry" es cleaner and easier to read than "entry in storage"

Reply:
Done!